base/sharedarray.jl

```diff
@@ -2,7 +2,21 @@
 
 import .Serializer: serialize_cycle_header, serialize_type, writetag, UNDEFREF_TAG
 
+const SA_CTR = Ref(1)
+next_sa_ctr() = (id = SA_CTR[]; SA_CTR[] = id+1; id)
+
+mutable struct SA_ID
```

Comment:
Any particular reason to make this `mutable`? I don't see any code that modifies one after construction. You wouldn't need the `hash` and `==` defs if it were just a `struct`.